src/flows/components/header/index.tsx

```diff
@@ -97,54 +88,44 @@ const FlowHeader: FC = () => {
     }
   }
 
-  const showShare = () => {
-    setSharing(true)
-    setLoadingToken(RemoteDataState.Loading)
-    getAuthorizations({query: {orgID}}).then(resp => {
-      if (resp.status !== 200) {
-        return
-      }
-
-      setLoadingToken(RemoteDataState.Done)
-      const _tokens = resp.data.authorizations.map(a => ({
-        token: a.token,
-        description: a.description || 'Describe this token',
-      }))
-
-      setTokens(_tokens)
-      event('Notebook share tokens', {count: _tokens.length})
-    })
-    event('Show Share Menu', {share: !!share ? 'sharing' : 'not sharing'})
-  }
-
   const hideShare = () => {
     setSharing(false)
-    setToken(null)
-    setLoadingToken(RemoteDataState.NotStarted)
   }
 
   const deleteShare = () => {
+    setLinkDeleting(RemoteDataState.Loading)
     deleteNotebooksShare({id: share.id})
       .then(() => {
+        setLinkDeleting(RemoteDataState.Done)
         hideShare()
         setShare(null)
         event('Delete Share Link')
       })
-      .catch(err => console.error('failed to delete share', err))
+      .catch(err => {
+        setLinkDeleting(RemoteDataState.Error)
+        console.error('failed to delete share', err)
+      })
   }
 
   const generateLink = () => {
+    event('Show Share Menu', {share: !!share ? 'sharing' : 'not sharing'})
+
+    if (!!share) {
+      setSharing(true)
```

Comment:
this is needed as you won't be able to see if the share already exists. If you are alluding to automatically showing the link if the share already exists, then it will always show the share link by default(if share exists). Is that what we want?